lib/plugins/bodyParser.js

```diff
@@ -160,25 +160,34 @@ function bodyParser(options) {
         }
 
         var parser;
-        var type = req.contentType().toLowerCase();
+        //var type = req.contentType().toLowerCase();
+        var type = req
+            .contentType()
+            .toLowerCase()
+            .split(';')[0];
 
-        switch (type) {
-            case 'application/json':
+        var jsonPatternMatcher = new RegExp('^application/[a-zA-Z.]+\\+json');
+
+        switch (true) {
```

Comment:
Switching on a constant is kind of uncommon - I think I would prefer we do something like map any `+json` type into application/json (if we find a match) and keep the existing switch statement as is.